core/deliverservice/blocksprovider/blocksprovider.go

```diff
@@ -159,14 +162,30 @@ func (b *blocksProviderImpl) DeliverBlocks() {
 
 			marshaledBlock, err := proto.Marshal(t.Block)
 			if err != nil {
-				logger.Errorf("[%s] Error serializing block with sequence number %d, due to %s", b.chainID, blockNum, err)
+				logger.Errorf("[%s] Error serializing block with sequence number %d, due to %s; Disconnecting client from orderer.", b.chainID, blockNum, err)
```

Comment:
we really don't need this.

Reply:
If there is a marshalling error, the right thing to do is to disconnect, in my opinion. Without it there will be a deadlock even if the orderer provides correct blocks past the corrupt one.